### ISSUE TITLE
TextArea: Replace aria-labelledby with aria-describedby

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2261]
Right now, aria-labelledby points to the description (errorMessage, helperMessage, character counter), which can cause:

The <label for="..."> not to contribute to the accessible name
The <textarea> to lose its proper connection to the label
Tests in Testing Library to fail with a non-labellable error

What if we make it like the Input component: the label provides the accessible name, and any additional text (error, helper message, character counter) is added via aria-describedby?

AC: Replace aria-labelledby with aria-describedby.

**Comment by olehnoskov-cengage** [comment]
**Tested on:**
- V4:
Docs: https://next--upbeat-sinoussi-f675aa.netlify.app/api/nav-tabs/#scrollable
Storybook: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/

- V5: 
Docs: https://rc--upbeat-sinoussi-f675aa.netlify.app/api/nav-tabs/#scrollable 
Storybook: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/buttongroup--default

Everything works as expected. No bugs found.
